pages/api/user/index.ts

```diff
@@ -0,0 +1,34 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import nextConnect from "next-connect";
+import prisma from "prisma/db";
+
+// import { generateRandomString, slugifyString } from "utils/shared/strings";
+import { getLoginId } from "utils/server/auth/user";
+
+export default nextConnect<NextApiRequest, NextApiResponse>()
+  .put(async (req, res) => {
+  	const loginId = await getLoginId(req);
+  	if (!loginId) {
+  		return res.status(403).json({ ok: false });
+  	}
+
+  	const { id, name, about, nameSlug, avatar } = req.body;
```

Comment:
We need to `slugifyString` here for `nameSlug` as we do in `/api/community/index`. Otherwise it will let me persist a slug with invalid characters.

Reply:
chose to slugify in the frontend since we do redirects after the renaming. also form validity checking which we can do later is in frontend too.